### ISSUE TITLE
Integrate continuous integration and coverage services.

### ISSUE THREAD

**Issue by LexiconCode** [dictation-toolbox/Caster#274]
[Travis CI](https://travis-ci.org/) is a hosted, distributed continuous integration service used to build and test software projects hosted at GitHub. Open source projects may be tested at no charge.

Travis CI will provide the following.
- It will automatically on every pull request run unit tests and check for have a passing or failing build.
- Once castor is a package. Travis CI will automatically publish stable releases from the master branch to pip the repository. 

[Codecov ](https://codecov.io/) focuses on integration and promoting healthy pull requests which will support Travis CI. 

Simply put, code coverage is a measurement used to express the degree to which the source code is executed when a test suite runs. A program with higher code coverage means that the test suite has had more of the source code executed when it runs. Thus, it implies that the source code has lesser chances of containing undetected bugs. 

Both of these services are free for open source projects :)

**Comment by LexiconCode** [comment]
@comodoro thanks for pointing that out. Fortunately we can [mocking](https://medium.com/@yeraydiazdiaz/what-the-mock-cheatsheet-mocking-in-python-6a71db997832) instead. 

An example of mocking imports from dragonfly's documentation config script.
https://github.com/Danesprite/dragonfly/blob/master/documentation/conf.py#L23

**Comment by comodoro** [comment]
I cannot seem to be able to get tests working without a recognition engine. Running e.g. as `python -c "import caster.lib.tests.testrunner; testrunner.run_tests()"` attempts to load apps and starts WSR and fails.

**Comment by LexiconCode** [comment]
@comodoro Yes and a workaround is being created through Dragonfly. See https://github.com/Danesprite/dragonfly/issues/36
